### PR TITLE
#162199433 Create featured-article routes

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
Create select-hero-article route

#### Description of Task to be completed?
- Create route for admin to select the featured article
- Create logic for auto-selection of featured article for admin
- Create route for fetching the featured article.
- Write tests for the routes
- Document the routes

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on Postman
- run `npm run start-dev`
- Add the title, body and description as `x-www-form-urlencoded`
- On postman run:
 PUT
 `localhost:5000/api/v1/admin/articles/featured` (set JWT token as`x-access-token` in header) to select a featured article
GET
`localhost:5000/api/v1/articles/featured` to fetch a featured article

NB: You need admin rights to access the PUT route.

#### Any background context you want to provide?
This route is required for selecting the article that will be displayed on the hero-image of the landing page.
The article can be auto-selected and it can be selected by the admin. Either way, the admin has to initiate the selection through this route.

#### What are the relevant pivotal tracker stories?
#162199433

#### Screenshots (if appropriate)

Admin-select:
screen shot 2018-11-27 at 1 00 10 pm

Auto-select:
screen shot 2018-11-27 at 1 00 21 pm

### CODE REVIEW

File: server/controllers/article.js [hunk 1]
```diff
@@ -219,6 +220,70 @@ class Article {
...
+    const benchmark = Math.ceil(allArticles.length * 0.05);
+
+    const topArticles = [];
+    for (let i = 0; i <= benchmark; i += 1) {
+      if (allArticles[i].dataValues.rank !== Infinity
+        || allArticles[i].dataValues.isDeleted !== true) {
+        topArticles.push(allArticles[i].dataValues);
+      }
+    }
+    const randomIndex = Math.floor(Math.random() * topArticles.length);
```
Comment (victor-nweze-deel) [inline server/controllers/article.js:272]: nice

**Review by victor-nweze-deel — COMMENTED** [review]
Good Job @TheoOkafor

**Comment by ucheg6** [comment]
Good job @TheoOkafor. 👍

File: server/controllers/article.js (hunk 1 above)
Comment (TheoOkafor) [reply server/controllers/article.js:272]: Thanks

**Comment by TheoOkafor** [comment]
> Good job @TheoOkafor. 👍

Thank you @ucheg6

**Review by seyi-adeleke — APPROVED** [review]

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**